tee-worker/ts-tests/integration-tests/data-providers.test.ts

```diff
@@ -0,0 +1,140 @@
+import { KeyringPair } from '@polkadot/keyring/types';
+import { u8aToHex } from '@polkadot/util';
+import { vip3CredentialJson, CredentialDefinition } from './common/credential-json';
+import colors from 'colors';
+describe('Test Vc (direct invocation)', function () {
+    let context: IntegrationTestContext = undefined as any;
+    let teeShieldingKey: KeyObject = undefined as any;
+    const substrateIdentities: CorePrimitivesIdentity[] = [];
+
+    const clientDir = process.env.LITENTRY_CLI_DIR;
+    const reqExtHash = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    const keyringPairs: KeyringPair[] = [];
+    let argvId = '';
+    this.timeout(6000000);
+    before(async () => {
+        context = await initIntegrationTestContext(process.env.WORKER_ENDPOINT!, process.env.NODE_ENDPOINT!);
+        teeShieldingKey = await getTeeShieldingKey(context);
+    });
+
+    // usage example:
+    // `pnpm run test-data-providers:local --id=vip3-membership-card-gold` for single test
+    // `pnpm run test-data-providers:local` for all tests
+    const argv = process.argv.indexOf('--id');
+    argvId = process.argv[argv + 1];
+    const {
+        protocol: workerProtocal,
+        hostname: workerHostname,
+        port: workerPort,
+    } = new URL(process.env.WORKER_ENDPOINT!);
+    const { protocol: nodeProtocal, hostname: nodeHostname, port: nodePort } = new URL(process.env.NODE_ENDPOINT!);
+
+    async function linkIdentityViaCli(id: string) {
+        const credentialDefinitions = vip3CredentialJson.find((item) => item.id === id) as CredentialDefinition;
+
+        const keyringPair = randomSubstrateWallet();
+        keyringPairs.push(keyringPair);
+        const formatAddress = u8aToHex(keyringPair.publicKey);
+
+        const substrateIdentity = (await buildIdentityFromKeypair(
+            new PolkadotSigner(keyringPair),
+            context
+        )) as CorePrimitivesIdentity;
+        substrateIdentities.push(substrateIdentity);
+        const eventsPromise = subscribeToEventsWithExtHash(reqExtHash, context);
+        try {
+            // CLIENT = "$CLIENT_BIN -p $NPORT -P $WORKER1PORT -u $NODEURL -U $WORKER1URL"
+            const commandPromise = zx`${clientDir} -p ${nodePort} -P ${workerPort} -u ${
+                nodeProtocal + nodeHostname
+            } -U ${workerProtocal + workerHostname}\
+                  trusted -d link-identity did:litentry:substrate:${formatAddress}\
+                  did:${credentialDefinitions.mockDid}\
+                  ${credentialDefinitions.mockWeb3Network}`;
+
+            await commandPromise;
+        } catch (error: any) {
+            console.log(`Exit code: ${error.exitCode}`);
+            console.log(`Error: ${error.stderr}`);
+            throw error;
+        }
+
+        const events = (await eventsPromise).map(({ event }) => event);
+        assert.equal(events.length, 1);
+    }
+
+    async function requestVc(id: string, index: number) {
```

Comment:
is it possible to utilise this function for other scenarios, like NFT holder or BNB domain?

Reply:
You're referring to other cases, right? Currently, this one is for VIP3, but I am testing [the other cases](https://github.com/litentry/litentry-parachain/pull/2434). To avoid ambiguity, I am redefining credentials array [here](https://github.com/litentry/litentry-parachain/pull/2431/files#diff-29be9a24deec6c20fbd8de209e73ae83c14eaa95a86baeb38dfabb9161c8f3b2R30).

Reply:
You're referring to other cases, right? Currently, this one is for VIP3, but I am testing [the other cases](https://github.com/litentry/litentry-parachain/pull/2434).